src/adapters/azure/index.ts

```diff
@@ -7,22 +7,29 @@ import { getHandleUpload } from './handleUpload'
 import { extendWebpackConfig } from './webpack'
 
 export interface Args {
-  connectionString: string
+  connectionString?: string
   containerName: string
   baseURL: string
   allowContainerCreate: boolean
+  storageClient?: ContainerClient
 }
 
 export const azureBlobStorageAdapter = ({
   connectionString,
   allowContainerCreate,
   containerName,
   baseURL,
+  storageClient: customStorageClient,
 }: Args): Adapter => {
-  let storageClient: ContainerClient | null = null
+  let storageClient: ContainerClient | null = customStorageClient ?? null
+
+  if (!connectionString && !customStorageClient) {
+    throw new Error('You must provide a connection string or a storage client')
+  }
+
   const getStorageClient = () => {
     if (storageClient) return storageClient
-    const blobServiceClient = BlobServiceClient.fromConnectionString(connectionString)
+    const blobServiceClient = BlobServiceClient.fromConnectionString(connectionString!) // We know this is defined if storageClient is not
```

Comment:
Comment is redundant with the type change suggested above.

Reply:
Makes sense, anyway I needed to add a simple cast here otherwise TS wasn't understanding the infered type, I could name each of the type and cast the args here, but that would require a slightly bigger refactor, a cast here isn't going to cause any harm.